js/tji-datasets-explore.js

```diff
@@ -220,31 +220,51 @@ TJIChartView.prototype.column_whitelist = [
 // of the charts, filter panel, etc.
 TJIChartView.prototype.get_data = function() {
   var that = this;
-  jQuery.getJSON('/cleaned_custodial_death_reports.json')
+  var url = '/cdr_compressed.json';
+  jQuery.getJSON(url)
     .done(function(data){
       that.state.data = data;
+      that.decompress_data();
       that.parse_data();
       that.create_charts();
       that.create_filter_panel();
       that.attach_events();
     })
-    .fail(function(e){
-      console.log('error fetching data from: ' + this.state.url, e);
+    .fail(function(jqxhr, textStatus, error){
+      console.log('error fetching data from: ' + url, error);
     })
 }
 
+// The data is compressed into a small json object for fast page loading,
+// but we expand it here for convenient manipulation in this app.
+TJIChartView.prototype.decompress_data = function() {
```

Comment:
📦 👍  🎉 
just some picky readability feedback that you can take or leave.
Could we put the column lookups in meta ... so what gets kicked back is
```
{
  meta: {
    lookups: {
      column_name: {},
      other_column_name: {},
    },
    num_records: 75468635792
  },
  records: {
    column_name: [],
    other_column_name: [],
  },
}
```
then 
```
var that = this;
var meta = data.meta;
var records = data.records;
var new_data = _.fill(Array(meta.num_records), {});
_.forEach(records, function(values, column) {
    var lookup  = meta.lookup[column] || {};
    _.each(values, function(v, idx) {
        new_data[idx][column] = (v === undefined || v === null || v === '') 
            ? that.missing_data_label 
            : (lookup[v] || v);
    }
});
this.state.data = new_data;
```
you could take care of handling missing values too with one loop through.

Reply:
Though I may be misunderstanding how the data is formatted.

Reply:
I think I like this new format. I've made the changes. Couple thoughts though:

1) I'd rather have the 'decompress' method *only* decompress, not do blank-value fills. It's a bit misleading to do non-decompression things there. So I'll keep it in the parse_data method I think? I just renamed it to `transform_data` since I think that's more honest.

2) It seems `_.fill(Array(...), {});` will fill every element with the *same* object, so they all edit each other. E.g.:
```
stuff = _.fill(Array(2), {})
stuff[0]['foo'] = 'bar'
stuff[1]['foo']
> 'bar'
```
I changed it to use `_.times` instead of a plain loop, but let me know if you have a more succinct way you prefer.